added feature and test for zipcode validation

xxKeefer opened the pull request:
# Description

Added feature and test for validating zip codes to utils

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ X] New feature (non-breaking change which adds functionality)


# Checklist:

- [X] I checked that there's no dataset update (can be done by running `npm run update-dataset`)
- [X] `npm test` passes on my machine
- [X] `npm run lint` passes on my machine

xxKeefer committed:
added feature and test for zipcode validation

Fcmam5 commented on src/utils/isValidZipCode.js:
```diff
@@ -0,0 +1,11 @@
+ * @returns {Boolean}
+ */
+const isValidZipCode = (code) =>
+  Number.isInteger(code) && code >= 1000 && code <= ZIP_COUNT;
```
Your point in #24 is valid IMO, we can pass a string like `'01027'`, so I'd suggest either just checking if code is not `NaN` (with [Number.isNaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN)), or you use `parseInt(code)` then you check for `parsedCode >= 1000 && parsedCode <= ZIP_COUNT;`

InternetRamen commented:
What should I do? I'm new to open source.

Fcmam5 commented:
> What should I do? I'm new to open source.

You can help us here by reviewing the code, and suggest any improvements. But first, we should wait for the PR author to publish the latest changes and then when the review is requested we can check it.
Or sometimes we notice something that we can improve and it could be outside of this PR scope, you can resolve it in a separate PR.

But for now, everything is fine :) just check our latest issue, it should be a nice welcoming-challenge for your open-source journey 🙂 !

xxKeefer committed:
updated isValidZipCode to handle valid strings.

xxKeefer commented:
# Description
Added feature and test for validating zip codes to utils
function now handles valid strings that can be parsed into into a valid integer, updated test to reflect this.

Fixes #26

# Type of change
Please delete options that are not relevant.

[ X] New feature (non-breaking change which adds functionality)

# Checklist:
[ X] I checked that there's no dataset update (can be done by running npm run update-dataset)
[ X] npm test passes on my machine
[X ] npm run lint passes on my machine

Fcmam5 commented on src/utils/isValidZipCode.js:
```diff
@@ -0,0 +1,18 @@
+const isValidZipCode = (code) => {
+  const parsedCode = parseInt(code, 10);
+  return (
+    !Number.isNaN(parsedCode) &&
```
We can remove this, as if the number is `NaN` our `Number.isInteger(parsedCode)` will return false
```suggestion
```

Fcmam5 commented on src/utils/isValidZipCode.js:
```diff
@@ -0,0 +1,18 @@
+const ZIP_COUNT = 48073;
+
+/**
+ * Check if a given integer is a valid zip code (between 1000 and 48073)
+ * @param {Number} code zip code
```
This function still work if we pass '01000' 

```suggestion
 * Check if a given parameter is a valid zip code (between 1000 and 48073)
 * @param {Number|String} code zip code
```

Fcmam5 reviewed (changes requested):
Clean! Thank you very much 😄 

I only have the follow suggestions. 
+ please export this file in `index.js` and document it in `README.md`

xxKeefer committed:
made requested changes and documentation

xxKeefer commented:
OK, should be good now, can you please look over the documentation in the README, i tried to follow the format but i am not sure if i matched it correctly.

Fcmam5 reviewed (approved):
Awesome! Thank you very much @xxKeefer for your contribution.

Fcmam5 commented:
@all-contributors add @xxKeefer for code
